rpc/cloudagent/compute/baremetalhost/moc_cloudagent_baremetalhost.proto

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache v2.0 license.
+
+syntax = "proto3";
+option go_package = "github.com/microsoft/moc/rpc/cloudagent/compute";
+package moc.cloudagent.compute;
+
+import "google/protobuf/wrappers.proto";
+import "moc_common_common.proto";
+import "moc_cloudagent_virtualmachine.proto";
+
+enum BareMetalHostPowerState {
+	Unknown = 0;
+	Running = 1 ;
+	Off = 2;
+}
+
+message BareMetalHostRequest {
+	repeated BareMetalHost BareMetalHosts = 1;
+	Operation OperationType = 2;
+}
+
+message BareMetalHostResponse {
+	repeated BareMetalHost BareMetalHosts = 1;
+	google.protobuf.BoolValue Result = 2;
+	string  Error = 3;
+}
+
+message BareMetalHostDisk {
+	string diskName = 1;
+	int32 diskSizeGB = 2;
+}
+
+message BareMetalHostStorageConfiguration {
+	repeated BareMetalHostDisk disks = 1;
+}
+
+message BareMetalHostOperatingSystemConfiguration {
+	string computerName = 1;
+	UserConfiguration administrator = 2;
+	repeated UserConfiguration users = 3;
+	repeated SSHPublicKey publicKeys = 5;
+	LinuxConfiguration linuxConfiguration = 6;
+}
+
+message BareMetalHostNetworkInterface {
+	string networkInterfaceName = 1;
+}
+
+message BareMetalHostNetworkConfiguration {
+	repeated BareMetalHostNetworkInterface interfaces = 1;
+}
+
+message BareMetalHostSize {
+	int32 cpuCount = 1;
+	int32 gpuCount = 2;
+	int32 memoryMB = 3;
+}
+
+message BareMetalHostHardwareConfiguration {
+	BareMetalHostSize machineSize = 1;
+}
+
+message BareMetalHost {
+	string name = 1;
+	string id = 2;
+	BareMetalHostStorageConfiguration storage = 3;
+	BareMetalHostOperatingSystemConfiguration os = 4;
```

Comment:
Curious, why do we keep OS config on BMH, as opposed to just having it on BMM?

Reply:
I had a similar thought, but the reason I kept it was primarily for the computerName field which is part of the OS config as we originally copied from Virtual Machine API.

Reply:
Why do we need to keep computerName on the BMH? Isn't that a property of the OS, hence somewhat transient?

Reply:
I thought a BMH could have a computerName too. Like the computerName when the provisioning OS is running on the bare metal prior to BMM is created.. But I will remove this.

Reply:
I dont see a usecase for that, let me know if you do. What would that computername be useful for? I dont think we would use it for IP resolution.

Reply:
Removed it.